include/aero/array/c_array.hpp

```diff
@@ -0,0 +1,49 @@
+#ifndef AERO_C_ARRAY_HPP
+#define AERO_C_ARRAY_HPP
+
+#include <aero/array/array.hpp>
+
+namespace aero {
+
+/// This Array subclass allows access to arrays implemented in C
+class CArray: public Array {
+public:
+  /// Default constructor
+  CArray() = delete;
+  /// Copy constructor
+  CArray(const CArray& other);
+  /// Move constructor
+  CArray(CArray&& other);
+  /// Construct from a pointer to a C-backed Array
+  CArray(void *fortran_array);
+
+  /// Destructor
+  ~CArray() override;
+
+  /// Copy assignment operator
+  CArray& operator=(CArray& other);
+  /// Assignment from a vector of Reals
+  CArray& operator=(const std::vector<Real> &values);
+  /// Default move assignment operator
+  CArray& operator=(CArray&&);
+
+  /// Returns a clone (deep-copy) of an array
+  Array* clone() const;
+  /// Copies data into an Array
+  void copy_in(const Real *input) override;
+  /// Copies data into an Array
+  void copy_in(const std::vector<Real> &input) override;
```

Comment:
This and its `copy_out` counterpart can be eliminated if we do it in the base class.